rest_framework/validators.py

```diff
@@ -211,7 +221,9 @@ def __call__(self, attrs):
         queryset = self.exclude_current_instance(attrs, queryset)
         if queryset.exists():
             message = self.message.format(date_field=self.date_field)
-            raise ValidationError({self.field: message})
+            raise ValidationError({
+                self.field: ValidationErrorMessage(message, code='unique'),
```

Comment:
Not 100% convinced by the new `ValidationErrorMessage` and the change in the error argument for nested cases here. It ends up being a bit odd because in the nested case these's a different way of indicating the error code, than is used in the simple case.

Alternatives might include (1) constraining the `code` argument to only support simple cases, or (2) support `code` as a nested structure.

We might need to do some more thinking in this area. Unsure.